src/services/keycloakService.ts

```diff
@@ -98,8 +99,11 @@ class KeycloakService {
    */
   public isAuthenticated(): boolean {
     this.checkKeycloakAvailability();
+    const authenticated = this.keycloakAuth.authenticated!;
 
-    return this.keycloakAuth.authenticated!;
+    authBroadcastService.send(authenticated);
```

Comment:
I suppose this might cause too much re-render of the `AppLayout` each time `isAuthenticated` is queried, which is not scarce.

If this would be achievable in the `AppLayout`:
```js
useEffect(() => {
    // update the user state
  }, [location.pathname]);
```

Reply:
Context API is probably not usable here, because it cannot be connected to the Keycloak code, if the Keyclode JS library does not provide event listeners.

Reply:
Broadcasting sounds like a preferable option, as one of its main advantages is that it can be used to detect user actions in other tabs within a same origin.

Maybe some nice feature could be to send user object directly instead of `authenticated` (boolean), so that whole user object would be directly automatically accessible via argument everywhere. When user is not available, `null` could be passed instead.